blaze/server/server.py

```diff
@@ -420,11 +493,29 @@ def compserver(payload, serial):
             500,
         )
 
-    return serial.dumps({
+    response = {
         'datashape': pprint(expr.dshape, width=0),
-        'data': result,
+        'data': serial.data_dumps(result),
         'names': expr.fields
-    })
+    }
+    if profiling:
+        import marshal
+        from pstats import Stats
+
+        profiler.disable()
+        if profiler_output == ':response':
+            from pandas.compat import BytesIO
+            file = BytesIO()
+        else:
+            file = open(_prof_path(profiler_output, expr), 'wb')
+
+        with file:
+            marshal.dump(Stats(profiler).stats, file)
```

Comment:
I just looked at how `cProfile` writes this file. It didn't offer a way to pass a file like object so I just reimplemented the function here allowing us to give an in memory buffer or on disk file.